packages/HorizonUI/Sources/HorizonUI/Sources/Components/Button/HorizonUI.ButtonStyles.swift

```diff
@@ -42,9 +53,40 @@ extension HorizonUI {
             self.fillsWidth = fillsWidth
             self.leading = leading
             self.trailing = trailing
+
+            self.badge = nil
+            self.badgeStyle = nil
+            self.icon = nil
+        }
+
+        fileprivate init(
+            background: any ShapeStyle,
+            foreground: Color,
+            badgeStyle: HorizonUI.Badge.Style,
+            isSmall: Bool = false,
+            icon: Image,
+            badge: String? = nil
+        ) {
+            self.background = AnyShapeStyle(background)
+            self.badge = badge
+            self.badgeStyle = badgeStyle
+            self.foreground = foreground
+            self.icon = icon
+            self.isSmall = isSmall
+
+            self.fillsWidth = false
+            self.leading = nil
+            self.trailing = nil
         }
 
         func makeBody(configuration: Configuration) -> some View {
+            if icon != nil {
```

Comment:
Please change this function to be like this & please a void to use AnyView, apple documentation said that 

```
@ViewBuilder
        func makeBody(configuration: Configuration) -> some View {
            if let icon {
                makeIconOnlyButtonType(icon: icon, configuration: configuration)
            }
             makePrimaryButtonType(configuration: configuration)
        }
```